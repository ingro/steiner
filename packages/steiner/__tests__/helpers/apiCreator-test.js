import createApi from '../../src/helpers/apiCreator';

function dummyClient(props) {
    return props;
}

// TODO: write tests about custom paramsMap and buildParams function
describe('apiCreator', () => {
    it('creates the correct api calls', () => {
        const apis = createApi('posts', dummyClient);

        expect(apis.fetch(42)).toMatchSnapshot();
        expect(apis.list()).toMatchSnapshot();
        expect(apis.create({ foo: 'bar' })).toMatchSnapshot();
        expect(apis.update(42, { foo: 'bar' })).toMatchSnapshot();
        expect(apis.delete(42)).toMatchSnapshot();
    });
});
