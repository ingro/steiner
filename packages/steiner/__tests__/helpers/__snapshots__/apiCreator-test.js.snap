exports[`apiCreator creates the correct api calls 1`] = `
Object {
  "url": "/posts/42",
}
`;

exports[`apiCreator creates the correct api calls 2`] = `
Object {
  "params": Object {
    "_limit": undefined,
    "_order": undefined,
    "_page": undefined,
    "_sort": undefined,
    "q": undefined,
  },
  "url": "/posts",
}
`;

exports[`apiCreator creates the correct api calls 3`] = `
Object {
  "data": Object {
    "foo": "bar",
  },
  "method": "post",
  "url": "/posts",
}
`;

exports[`apiCreator creates the correct api calls 4`] = `
Object {
  "data": Object {
    "foo": "bar",
  },
  "method": "patch",
  "url": "/posts/42",
}
`;

exports[`apiCreator creates the correct api calls 5`] = `
Object {
  "method": "delete",
  "url": "/posts/42",
}
`;
